core/unit_test/sycl/TestSYCL_InterOp_StreamsMultiGPU.cpp

```diff
@@ -0,0 +1,78 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <TestSYCL_Category.hpp>
+#include <TestMultiGPU.hpp>
+
+namespace {
+
+std::array<TEST_EXECSPACE, 2> get_execution_spaces() {
+  std::vector<sycl::device> gpu_devices =
+      sycl::device::get_devices(sycl::info::device_type::gpu);
+  auto exception_handler = [](sycl::exception_list exceptions) {
+    bool asynchronous_error = false;
+    for (std::exception_ptr const &e : exceptions) {
+      try {
+        std::rethrow_exception(e);
+      } catch (sycl::exception const &e) {
+        std::cerr << e.what() << '\n';
+        asynchronous_error = true;
+      }
+    }
+    if (asynchronous_error)
+      Kokkos::Impl::throw_runtime_exception(
+          "There was an asynchronous SYCL error!\n");
+  };
```

Comment:
Why are we specifying an asynchronous error handler?
Any good reason not to just use the constructor that takes a device and a property list?

Reply:
It's just the default we are using for the internal queues. It doesn't really matter here.

Reply:
Then drop it

Reply:
Here you go, see https://github.com/kokkos/kokkos/pull/6887/commits/0f04a5fca52a045a8f110255e05786f0e80ff798!